app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/miband/BatteryInfo.java

```diff
@@ -15,20 +23,45 @@ public int getLevelInPercent() {
         return 50; // actually unknown
     }
 
-    public String getStatus() {
+    public BatteryState getState() {
         if (mData.length >= 10) {
             int value = mData[9];
             switch (value) {
-                case 1:
-                    return GBApplication.getContext().getString(R.string.battery_low);
-                case 2:
-                    return GBApplication.getContext().getString(R.string.battery_medium);
-                case 3:
-                    return GBApplication.getContext().getString(R.string.battery_full);
-                case 4:
-                    return GBApplication.getContext().getString(R.string.battery_not_charging);
+                case DEVICE_BATTERY_NORMAL:
+                    return BatteryState.BATTERY_NORMAL;
+                case DEVICE_BATTERY_LOW:
+                    return BatteryState.BATTERY_LOW;
+                case DEVICE_BATTERY_CHARGING:
+                    return BatteryState.BATTERY_CHARGING;
+                case DEVICE_BATTERY_CHARGING_FULL:
+                    return BatteryState.BATTERY_CHARGING_FULL;
+                case DEVICE_BATTERY_CHARGE_OFF:
+                    return BatteryState.BATTERY_NOT_CHARGING_FULL;
             }
         }
-        return GBApplication.getContext().getString(R.string._unknown_);
+        return BatteryState.UNKNOWN;
+    }
+
+    public GregorianCalendar getLastChargeTime() {
```

Comment:
I guess I should enhance the documentation of the *Coordinator classes some more. Maybe they should even be renamed to something like MiBandUISupport or something like that. The coordinator classes are meant to be used in the UI part only, not by the service. And they are meant to be used generically through the DeviceCoordinator interface.

That's why I suggested to put this into the MiBandConst class. If this is poorly named, maybe we should find a better place for it as well.